flink-yarn/src/test/java/org/apache/flink/yarn/YarnResourceManagerDriverTest.java

```diff
@@ -669,4 +650,111 @@ void prepareForTestStartTaskExecutorProcessVariousSpec(
+                        "aborted exit code",
+                        ContainerExitStatus.ABORTED);
+        testingGetContainerCompletedCause(containerStatus);
+    }
+
+    @Test
+    public void testGetConatainerCompletedCauseForDiskFailed() throws Exception {
+        ContainerStatus containerStatus =
+                ContainerStatusPBImpl.newInstance(
+                        testingContainer.getId(),
+                        ContainerState.COMPLETE,
+                        "disk failed exit code",
+                        ContainerExitStatus.DISKS_FAILED);
+        testingGetContainerCompletedCause(containerStatus);
+    }
+
+    @Test
+    public void testGetConatainerCompletedCauseForUnknown() throws Exception {
+        ContainerStatus containerStatus =
+                ContainerStatusPBImpl.newInstance(
+                        testingContainer.getId(), ContainerState.COMPLETE, "unknown exit code", -1);
+        testingGetContainerCompletedCause(containerStatus);
+    }
+
+    public void testingGetContainerCompletedCause(ContainerStatus containerStatus)
+            throws Exception {
+        new Context() {
+            {
+                addContainerRequestFutures.add(new CompletableFuture<>());
+                addContainerRequestFutures.add(new CompletableFuture<>());
+
+                testingYarnAMRMClientAsyncBuilder.setAddContainerRequestConsumer(
+                        (ignored1, ignored2) ->
+                                addContainerRequestFutures
+                                        .get(
+                                                addContainerRequestFuturesNumCompleted
+                                                        .getAndIncrement())
+                                        .complete(null));
+                resourceEventHandlerBuilder.setOnWorkerTerminatedConsumer(
+                        (ignore1, ignore2) ->
+                                getDriver().requestResource(testingTaskExecutorProcessSpec));
+
+                runTest(
+                        () -> {
+                            runInMainThread(
+                                    () ->
+                                            getDriver()
+                                                    .requestResource(
+                                                            testingTaskExecutorProcessSpec));
+                            resourceManagerClientCallbackHandler.onContainersAllocated(
+                                    ImmutableList.of(testingContainer));
+                            ContainerStatus testingContainerStatus =
+                                    createTestingContainerCompletedStatus(containerStatus);
+
+                            assertTrue(
+                                    YarnResourceManagerDriver.getContainerCompletedCause(
+                                                    testingContainerStatus)
+                                            .contains(testingContainerStatus.getDiagnostics()));
```

Comment:
I would suggest to factor this out into its own test case. That way we have a test that checks the completed cause and another testing the container completed callback.